spec/goldiloader/goldiloader_spec.rb

```diff
@@ -748,6 +748,185 @@
+        expect(post.association(:tags)).not_to be_loaded
+      end
+    end
+
+    context "using enabled block" do
+      shared_examples "it auto eager loads association" do |subject, association|
+        specify do
+          Goldiloader.enabled do
+            items = send(subject)
+            # Sanity check that associations aren't loaded yet
+            items.each do |item|
+              expect(item.association(association)).not_to be_loaded
+            end
+
+            relation = items.first.association(association)
+            queries = { relation.klass => 1 }
+
+            if relation.options[:through]
+              through = items.first.association(relation.options[:through])
+              queries[through.klass] = 1
+            end
+
+            expect do
+              # Force load via auto include
+              rel = items.to_a.first.association(association)
+              # handle single and multi relations
+              rel.load_target
+
+              # Check that associations are loaded
+              items.each do |blog|
+                expect(blog.association(association)).to be_loaded
+              end
+            end.to execute_queries(queries)
+          end
+        end
+      end
+
+      let(:blogs) { Blog.order(:name).to_a }
+      let(:posts) { Post.order(:title).to_a }
+      let(:users) { User.order(:name).to_a }
+      let(:posts_through_blogs) { blogs.flat_map(&:posts) }
+      let(:tags) { Tag.where('parent_id IS NOT NULL').order(:name).to_a }
+      let(:users_through_tags) { tags.map(&:owner).select { |o| o.is_a?(User) } }
+
+      it_behaves_like "it auto eager loads association", :blogs, :posts
+      it_behaves_like "it auto eager loads association", :blogs, :authors
+      it_behaves_like "it auto eager loads association", :blogs, :posts_overridden
+      it_behaves_like "it auto eager loads association", :posts, :blog
+      it_behaves_like "it auto eager loads association", :users, :address
+      it_behaves_like "it auto eager loads association", :posts_through_blogs, :author
+      it_behaves_like "it auto eager loads association", :tags, :parent
+      it_behaves_like "it auto eager loads association", :users_through_tags, :posts
+
+      it "auto eager loads associations when the model is loaded via find" do
+        Goldiloader.enabled do
+          blog = Blog.find(blog1.id)
+
+          blog.posts.to_a.first.author
+          blog.posts.each do |post|
+            expect(post.association(:author)).to be_loaded
+          end
+        end
+      end
+
+      it "doesn't auto eager loads associations outside the block" do
+        blog = nil
+        Goldiloader.enabled do
+          blog = Blog.find(blog1.id)
+        end
+
+        arr = blog.posts.to_a
+        arr.first.author
+        arr.drop(1).each do |post|
+          expect(post.association(:author)).not_to be_loaded
+        end
+      end
+
+      it "sets inverse associations properly" do
```

Comment:
Is this testing anything that isn't covered by this [test](https://github.com/salsify/goldiloader/blob/master/spec/goldiloader/goldiloader_spec.rb#L220)? The `Goldiloader.enabled` block seems incidental to what's being tested.